src/zenml/integrations/label_studio/flavors/label_studio_annotator_flavor.py

```diff
@@ -30,18 +32,31 @@
 DEFAULT_LOCAL_LABEL_STUDIO_PORT = 8093
 
 
-class LabelStudioAnnotatorConfig(
-    BaseAnnotatorConfig, AuthenticationConfigMixin
-):
-    """Config for the Label Studio annotator.
+class LabelStudioAnnotatorSettings(BaseSettings):
+    """Label studio annotator settings.
 
     Attributes:
         instance_url: URL of the Label Studio instance.
         port: The port to use for the annotation interface.
+        api_key: The api_key for label studio.
     """
 
     instance_url: str = DEFAULT_LOCAL_INSTANCE_URL
-    port: int = DEFAULT_LOCAL_LABEL_STUDIO_PORT
+    port: Optional[int] = DEFAULT_LOCAL_LABEL_STUDIO_PORT
+    api_key: Optional[str] = SecretField()
```

Comment:
In most integrations (that use an API key or a port as a part of a flavor implementation), we placed these fields as parts of the `Config` class instead of the `Settings` class. Is there a reason why we opted out of this behavior? If not, I would propose that the `port` and `api_key` actually belong to the `Config` object and have to be defined upon the registration of the component.

Reply:
We did this specifically so as to allow quick switching for this component when 3 people were using the same annotator, but we didn't want to each separately to have to configure a separate component. We wanted to use the same component in our shared server, but with each passing in different values for the `api_key` via a config file.